concert/experiments/imaging.py

```diff
@@ -857,3 +861,216 @@ async def _take_radios(self):
+                                            num_steps_per_period=num_steps_per_period,
+                                            propagation_distance=propagation_distance)
+
+        self._remove_parameter('num_flats')
+        self._remove_parameter('num_projections')
+
+        reference_stepping = Acquisition("reference_stepping", self._take_reference_scan)
+        self.remove(self.get_acquisition("flats"))  # No flats required due to ref. stepping
+        self.add(reference_stepping)
+
+        object_stepping = Acquisition("object_stepping", self._take_object_scan)
+        self.remove(self.get_acquisition("radios"))  # No radios required due to obj. stepping
+        self.add(object_stepping)
+
+    async def _take_reference_scan(self):
+        """
+        Starts the sample exposure, moves the flat_motor in the *reference_position* and runs
+            self._take_scan().
+        """
+        await self.start_sample_exposure()
+        await self._flat_motor.set_position(await self.get_flat_position())
+        async for image in self._take_scan():
+            yield image
+
+    async def _take_object_scan(self):
+        """
+        Starts the sample exposure, moves the flat_motor in the *radio_position* and runs
+            self._take_scan().
+        """
+        await self.start_sample_exposure()
+        await self._flat_motor.set_position(await self.get_radio_position())
+        async for image in self._take_scan():
+            yield image
+
+    async def _take_scan(self):
+        """
+        Scans the stepping motor and acquires a frame after each position is reached.
+
+        As step *size grating_period* / *num_steps_per_period* is used.
+        A total of *num_steps_per_period* * *num_periods* frames is acquired.
+        """
+        step_size = await self.get_grating_period() / await self.get_num_steps_per_period()
+        if await self._camera.get_state() != "standby":
+            await self._camera.stop_recording()
+        await self._camera.set_trigger_source("SOFTWARE")
+        async with self._camera.recording():
+            for i in range(await self.get_num_periods() * await self.get_num_steps_per_period()):
+                await self._stepping_motor.set_position(i * step_size +
+                                                        await self.get_stepping_start_position())
+                try:
+                    await self._camera.trigger()
+                except:
+                    pass
+                yield await self._camera.grab()
+
+    async def finish(self):
+        """
+        This function calls stop_sample_exposure() at the end of the experiment run.
+        """
+        await self.stop_sample_exposure()
+
+    async def start_sample_exposure(self):
+        """
+        This function must implement in a way that the sample is exposed by radiation, like opening
+        a shutter or starting an X-ray tube.
+        """
+        raise NotImplementedError()
```

Comment:
In the manager PR you are raising without (), we should keep it consistent throughout concert.

Reply:
OK